thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/FunctionCreationOnboardingTask.java

```diff
@@ -179,27 +184,45 @@ public void run() {
-      if (Response.Status.OK.equals(response.getStatusInfo())) {
-        long functionId = Long.valueOf(response.getEntity().toString());
-        anomalyFunction = anoomalyFunctionDAO.findById(functionId);
-        executionContext.setExecutionResult(ANOMALY_FUNCTION_CONFIG, anomalyFunction);
-      } else {
-        throw new UnsupportedOperationException("Get Exception from Anomaly Function Creation End-Point");
+      anomalyFunction.setMetricId(metricConfig.getId());
+      anomalyFunction.setMetric(metricConfig.getName());
+      anomalyFunction.setTopicMetric(metricConfig.getName());
+      anomalyFunction.setMetrics(Arrays.asList(metricConfig.getName()));
+      anomalyFunction.setCollection(datasetConfig.getDataset());
+      anomalyFunction.setCron(configuration.getString(CRON_EXPRESSION, defaultFunctionSpec.getCron()));
+      anomalyFunction.setMetricFunction(MetricAggFunction.valueOf(
+          configuration.getString(METRIC_FUNCTION, defaultMetricFunction)));
+      String filters = configuration.getString(FILTERS);
+      if (!org.apache.commons.lang3.StringUtils.isBlank(filters)) {
+        filters = URLDecoder.decode(filters, DEFAULT_URL_DECODER);
+        String filterString = ThirdEyeUtils.getSortedFiltersFromJson(filters);
+        anomalyFunction.setFilters(filterString);
       }
+      if (org.apache.commons.lang3.StringUtils.isNotEmpty(configuration.getString(EXPLORE_DIMENSION))) {
+        anomalyFunction.setExploreDimensions(FunctionCreationUtils.getDimensions(datasetConfig, configuration.getString(EXPLORE_DIMENSION)));
+      }
+
+      anomalyFunction.setWindowSize(configuration.getInt(WINDOW_SIZE, defaultFunctionSpec.getWindowSize()));
+      anomalyFunction.setWindowUnit(TimeUnit.valueOf(configuration
+          .getString(WINDOW_UNIT, defaultFunctionSpec.getWindowUnit().name())));
+      anomalyFunction.setWindowDelay(configuration.getInt(WINDOW_DELAY, DEFAULT_WINDOW_DELAY));
+      anomalyFunction.setWindowDelayUnit(TimeUnit.valueOf(
+          configuration.getString(WINDOW_DELAY_UNIT, dataGranularity.getUnit().toString())));
+      anomalyFunction.setType(configuration.getString(FUNCTION_TYPE, defaultFunctionSpec.getType()));
+      anomalyFunction.setProperties(com.linkedin.thirdeye.datalayer.util.StringUtils.
+          encodeCompactedProperties(defaultFunctionProperties));
+      if (defaultFunctionSpec.getFrequency() != null) {
+        anomalyFunction.setFrequency(defaultFunctionSpec.getFrequency());
+      }
+      anomalyFunction.setBucketSize(dataGranularity.getSize());
+      anomalyFunction.setBucketUnit(dataGranularity.getUnit());
+      anomalyFunction.setIsActive(configuration.getBoolean(IS_ACTIVE, DEFAULT_IS_ACTIVE));
+
+      anoomalyFunctionDAO.update(anomalyFunction);
+
+      executionContext.setExecutionResult(ANOMALY_FUNCTION_CONFIG, anomalyFunction);
     } catch (Exception e) {
-      throw new IllegalArgumentException(e);
+      throw new IllegalArgumentException(e.fillInStackTrace());
```

Comment:
the original, nested exception passing, is the correct way of doing this

Reply:
Thanks for the comment. It's been fixed.